build/lint.js

```diff
@@ -38,13 +37,13 @@ function _spawn(proc, args, done) {
 }
 
 function _lintJS(files, done) {
-    _spawn('jshint', files, done);
+    _spawn((process.platform === 'win32') ? 'jshint.cmd' : 'jshint', files, done);
 }
 
 function _lintCSS(files, done) {
     var rules = JSON.parse(fs.readFileSync(_c.ROOT + ".csslintrc", "utf-8")),
         options = ["--errors=" + rules, "--format=compact", "--quiet"];
-    _spawn('csslint', files.concat(options), function (/*code*/) {
+    _spawn((process.platform === 'win32') ? 'csslint.cmd' : 'csslint', files.concat(options), function (/*code*/) {
```

Comment:
As far as I can tell this is OK.

I have msysgit installed.  I ran that “Git Bash” shortcut to start a new command box running the msgit shell.  In that box I ran “jake lint” and it worked fine.  I also ran “sh” in a command box and then ran jake lint and that worked too.

I think the reason this is OK is because build/lint.js spawns a subprocess to run the lint command.  The command you spawn doesn’t go through the unix-like shell, which is why the “.cmd” form works.

    Julian

From: Brent Lintner [mailto:notifications@github.com]
Sent: Sunday, April 19, 2015 2:50 PM
To: apache/incubator-ripple
Cc: Horn, Julian C
Subject: Re: [incubator-ripple] Made jake [lint, build, test] work on windows (#47)


In build/lint.js<https://github.com/apache/incubator-ripple/pull/47#discussion_r28655479>:

>  }

>

>  function _lintCSS(files, done) {

>      var rules = JSON.parse(fs.readFileSync(_c.ROOT + ".csslintrc", "utf-8")),

>          options = ["--errors=" + rules, "--format=compact", "--quiet"];

> -    _spawn('csslint', files.concat(options), function (/*code*/) {

> +    _spawn((process.platform === 'win32') ? 'csslint.cmd' : 'csslint', files.concat(options), function (/*code*/) {

Just wondering if the platform returned when using https://msysgit.github.io/ is, as this could break it in there (since jshint and csslint are as in osx/linux land, if so?

—
Reply to this email directly or view it on GitHub<https://github.com/apache/incubator-ripple/pull/47/files#r28655479>.